src/components/InngestFunction.ts

```diff
@@ -114,10 +115,10 @@ export class InngestFunction<Events extends Record<string, EventPayload>> {
   /**
    * Generate an ID based on the function's name.
    */
-  #generateId() {
+  #generateId(prefix?: string) {
     const join = "-";
 
-    return this.#opts.name
+    return `${prefix || ""}-${this.#opts.name}`
```

Comment:
Not that it's something that happens right now, but could this use `.join()` to omit a leading `-` if no `prefix` is passed?

```suggestion
    return [prefix || "", this.#opts.name].join("-");
```

Reply:
Aha, this is taken care of in your slugify regex which removes leading and trailing slashes.  I'll commit this, though, as it's safer and nicer!